### PR TITLE
  [FT-681] Transactional inconsistency with rollback

### PR DESCRIPTION
**Pull request by Jun-Yuan**
Remake the pull request
http://jenkins.percona.com/view/TokuDB/job/PerconaFT-param/36/

### CODE REVIEW

File: src/tests/rollback-inconsistency.cc [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        DBT val = { .data = str, .size = 220 };
+        r = db->put(db, txn, &key, &val, 0);
+        assert_zero(r);
+    }
+
+    r = txn->commit(txn, 0); assert_zero(r);
+    
+    r = parent->abort(parent); assert_zero(r);
+    //try to search anything in the lost range
+        int k = htonl(200);
```
Comment (christianrober) [inline src/tests/rollback-inconsistency.cc]: Why is this still indented?  Did you intend to perform more than one `db->get()` in a loop?

Comment (Jun-Yuan) [inline src/tests/rollback-inconsistency.cc]: Yeah. You are too smart. I was doing a loop before then I thought it is enough to do one quick lookup for proof-of-concept. Then I forgot I already deleted the loop...:(.............will remove the indent later today...in the middle of chasing read-commit.

File: src/tests/rollback-inconsistency.cc [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+    r = txn->commit(txn, 0); assert_zero(r);
+    //again eviction due to the memory pressure. 256 rows is the point when that rollback log spills out. The spilled node
+    //will be written back but will not be dirtied by including rollback nodes from child txn(in which case the bug is bypassed).
+ 
+    r = env->txn_begin(env, parent, &txn, 0); assert_zero(r);
+
+    // populate
+    for (int i = 256; i < 512; i++) {
+        int k = htonl(i);
```
Comment (christianrober) [inline src/tests/rollback-inconsistency.cc]: The contents of this "populate" loop look identical to the previous two loops, aside from the initial and ending conditions of `i`.  Perhaps they should be refactored into a function called `populate_in_range` or `put_keys_with_child_txn` that takes the parent TXN, the db and the two `i` bounds.  Then this function would be called three times, wrapped by the creation and aborting of the parent transaction, followed by the `get` step to verify the bug.  That would be a lot easier to read/understand and avoid code duplication:

```
DB_TXN *parent = NULL;
int r = env->txn_begin(env, NULL, &parent, 0); assert_zero(r);
populate(env, db, parent, 0, 128);
populate(env, db, parent, 128, 256);
populate(env, db, parent, 256, 512);
r = parent->abort(parent); assert_zero(r);
...
```

**Comment by Jun-Yuan** [comment]
OK. That sounds good. I will modify as the suggestion. Thank you.

**Pull request closed without merging**